Sources/AWSSDKSwiftCore/AWSClient.swift

```diff
@@ -301,7 +301,15 @@ extension AWSClient {
         switch serviceProtocol.type {
         case .json, .restjson:
             if let payload = Input.payloadPath, let payloadBody = mirror.getAttribute(forKey: payload.toSwiftVariableCase()) {
-                body = Body(anyValue: payloadBody)
+                switch payloadBody {
+                case is AWSShape:
+                    let inputBody: Body = .json(try AWSShapeEncoder().encodeToJSONUTF8Data(input))
```

Comment:
Ouch, yah this is ugly. I think this is just showing we need to have better data-transformation options available here, likely through `Codable` and proper encoders/decoders of the various formats, yah?

Reply:
This one is really tough b/c the object we need to encode is actually a variable on the Input instance who's name is dynamically defined by the `payload` string. To access this, @noppoMan implemented reflection with the Mirror class, but this returns an object of type `Any`, so `Codable` won't work. The compiler simply won't allow us to pass the object to the encoder unless it knows for sure that the dataType will always be a `Codable` object (which is doesn't since the type is `Any`).

Even if we add control flow that only calls the encoder if `payloadBody` is an `AWSShape`, it doesn't work b/c it's type is still `Any` as far as the compiler is concerned. 

Upcasting to an `AWSShape` doesn't work, b/c the encoder wants a Generic type as input, not a protocol.

So I don't have a much better solution that this really. Perhaps some super class that implements `AWSShape`, then every service shape extends this class... but seems like a lot to change to resolve this...